src/pushsource/_impl/model/ami.py

```diff
@@ -99,9 +99,33 @@ class AmiPushItem(VMIPushItem):
     )
     """Billing codes associated with this image."""
 
+    release_notes = attr.ib(type=str, default=None, validator=optional_str)
+    """"Notes regarding changes in the newest version."""
+
+    usage_instructions = attr.ib(type=str, default=None, validator=optional_str)
+    """Step by step instructions for the end-user to launch, configure and access the product. 
+
+    Write with a less-technical customer in mind.
+    """
+
+    recommended_instance_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Recommended instance type for AMI e.g. "t2.micro", "r6a.large" """
+
+    marketplace_entity_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Type of entity e.g. "AMIProduct", "SaasProduct", "ServerProduct". """
+
     image_id = attr.ib(type=str, default=None, validator=optional_str)
     """AMI Image ID used to reference the image in AWS."""
 
+    scanning_port = attr.ib(type=int, default=None, validator=instance_of(int))
+    """AMI scanning port, used when importing the AMI into AWS Marketplace to validate the AMI."""
+
+    user_name = attr.ib(type=str, default=None, validator=optional_str)
+    """The username used to login to the AMI. (Usually set to ec2-user)"""
+
+    security_groups = attr.ib(type=str, default=None, validator=optional_str)
+    """Automatically created security groups for the product. """
+
     @classmethod
     def _from_data(cls, data):
```

Comment:
I see now that it's not possible to load any AMI push items right now using the new fields (at any non-default value), because:

- right now, the only supported way to load AMIs is by `staged` backend
- that loads items via this `_from_data` method
- but none of the new fields are copied across here

So there is no way to actually load/test AmiPushItem with the new fields as of today.

I think your end goal is to make this work with `errata` backend, but it seems all the new fields ought to be supported when using `staged` backend as well.

I suggest to add handling for all the new fields here in `_from_data`.  Then, if you were to add some example values for the new fields into tests/staged/data/simple_ami/staged.yaml, the fields would be covered by an existing test.  Alternatively the simple_ami test could be copied and modified as a new test.

Reply:
Implemented this one.